Update on Landsat README

pholleway opened the pull request:
Related Issue(s): README file 

Description: Updated README to follow template and added table providing overview on Landsat Mission  sensors and band wavelengths.

PR checklist:

 [ ] Code is formatted (run scripts/format).
 [ ] Code lints properly (run scripts/lint).
 [ ] Tests pass (run scripts/test).
 [ ]  Documentation has been updated to reflect changes, if applicable.
 [ ] Changes are added to the [CHANGELOG](https://github.com/stactools-packages/modis/CHANGELOG.md).

gadomski reviewed (approved):
LGTM, @pjhartzell you mind reviewing as well? Thanks.

pjhartzell reviewed (approved):
Looks good.

gadomski committed:
Update on README

Added template language (a few fields still need to be filled out by others) and table providing an overview on Landsat Missions sensors and band wavelengths.
